lib/bundler/runtime.rb

```diff
@@ -37,7 +37,17 @@ def setup(*groups)
 
         Bundler.rubygems.mark_loaded(spec)
         load_paths = spec.load_paths.reject {|path| $LOAD_PATH.include?(path) }
-        $LOAD_PATH.unshift(*load_paths)
+
+        # See Gem::Specification#add_self_to_load_path (since RubyGems 1.8)
+        insert_index = Gem.load_path_insert_index
```

Comment:
We have a rubygems_integration.rb file where we wrap all interactions with `Gem`... Just so we have a centralized place to handle changes between RubyGems versions. If you create a method there to wrap this, you can call it from `Bundler.rubygems.some_method`.